libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -539,90 +434,62 @@ void ies_linalg_solve_S(const matrix_type *W0, const matrix_type *Y,
+                             subspace_dimension);
+
+        matrix_free(scaledE);
     } else if (ies_inversion == IES_INVERSION_SUBSPACE_EE_R) {
-        fprintf(log_fp,
-                "Subspace inversion using ensemble generated full R=EE. "
-                "(ies_inversion=%d)'\n",
-                ies_inversion);
-        matrix_scale(E, nsc);
         matrix_type *Et = matrix_alloc_transpose(E);
         matrix_type *Cee = matrix_alloc_matmul(E, Et);
         matrix_scale(
             Cee,
-            nsc *
-                nsc); // since enkf_linalg_lowrankCinv solves (SS' + (N-1) R)^{-1}
-        if (dbg)
-            matrix_pretty_fprint_submat(Cee, "Cee", "%11.5f", log_fp, 0,
-                                        m_nrobs, 0, m_nrobs);
-        enkf_linalg_lowrankCinv(S, Cee, X1, eig, truncation,
+            1.0 /
+                ((ens_size - 1) *
+                 (ens_size -
+                  1))); // since enkf_linalg_lowrankCinv solves (SS' + (N-1) R)^{-1}
+
+        enkf_linalg_lowrankCinv(S, Cee, X1, eig.data(), truncation,
                                 subspace_dimension);
+
         matrix_free(Et);
         matrix_free(Cee);
     } else if (ies_inversion == IES_INVERSION_SUBSPACE_EXACT_R) {
-        fprintf(log_fp,
-                "Subspace inversion using 'exact' full R. (ies_inversion=%d)\n",
-                ies_inversion);
-        matrix_scale(
-            R,
-            nsc *
-                nsc); // since enkf_linalg_lowrankCinv solves (SS' + (N-1) R)^{-1}
-        if (dbg)
-            matrix_pretty_fprint_submat(R, "R", "%11.5f", log_fp, 0, m_nrobs, 0,
-                                        m_nrobs);
-        enkf_linalg_lowrankCinv(S, R, X1, eig, truncation, subspace_dimension);
-    }
-
-    int nrsing = 0;
-    fprintf(log_fp, "\nEig:\n");
-    for (int i = 0; i < nrmin; i++) {
-        fprintf(log_fp, " %f ", eig[i]);
-        if ((i + 1) % 20 == 0)
-            fprintf(log_fp, "\n");
-        if (eig[i] < 1.0)
-            nrsing += 1;
+        matrix_type *scaledR = matrix_alloc_copy(R);
+        matrix_scale(scaledR, nsc * nsc);
+        // std_enkf.cpp: 165
+        enkf_linalg_lowrankCinv(S, scaledR, X1, eig.data(), truncation,
+                                subspace_dimension);
+        matrix_free(scaledR);
     }
-    fprintf(log_fp, "\n");
-
-    /*    X3 = X1 * diag(eig) * X1' * H (Similar to Eq. 14.31, Evensen (2007))                                  */
-    enkf_linalg_genX3(X3, X1, H, eig);
 
-    if (dbg)
-        matrix_pretty_fprint_submat(X1, "X1", "%11.5f", log_fp, 0, m_nrobs, 0,
-                                    std::min(m_nrobs, nrmin - 1));
-    if (dbg)
-        matrix_pretty_fprint_submat(X3, "X3", "%11.5f", log_fp, 0, m_nrobs, 0,
-                                    m_ens_size);
+    {
+        /*    X3 = X1 * diag(eig) * X1' * H (Similar to Eq. 14.31, Evensen (2007))                                  */
```

Comment:
This comment wraps on github so it would be great if it were shorter. Could probably just remove some redundant space before X3.

Reply:
OK